### PR TITLE
remove irradiance parameter from ModelChain.run_model and ModelChain.prepare_inputs

### LINKED ISSUE #488: remove irradiance parameter from ModelChain.run_model
#239 deprecated the irradiance parameter in ``ModelChain.run_model``. It's time to remove it.

### PR DESCRIPTION
**Pull request by alamathe1**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python!

You may submit a pull request with your code at any stage of completion, however, before the code can be merged the following items must be addressed:

 - [x] Closes #488
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

Please don't hesitate to ask for help if you're unsure of how to accomplish any of the above. You may delete all of these instructions except for the list above.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

### CODE REVIEW

File: pvlib/modelchain.py [hunk 1]
```diff
@@ -754,10 +752,6 @@ def prepare_inputs(self, times=None, irradiance=None, weather=None):
                    "Please use the weather parameter to pass a DataFrame " +
                    "with irradiance (ghi, dni, dhi), wind speed and " +
                    "temp_air.\n")
-        if irradiance is not None:
```
Comment (wholmgren) [inline pvlib/modelchain.py]: The lines from "# The following part could be removed together with the irradiance" through "# **** End ****" should also be deleted.

**Comment by wholmgren** [comment]
Thanks @alamathe1! I'll merge this in a few days if there are no other comments. The test failures are unrelated.

**Comment by alamathe1** [comment]
@wholmgren Alright thanks!

**Comment by wholmgren** [comment]
Thanks again @alamathe1!

**Pull request merged**